crates/goose-server/src/routes/recipe.rs

```diff
@@ -350,6 +350,13 @@ async fn save_recipe(
 ) -> Result<Json<SaveRecipeResponse>, ErrorResponse> {
     let Json(raw_json) = payload.map_err(json_rejection_to_error_response)?;
     let request = deserialize_save_recipe_request(raw_json)?;
+    let has_security_warnings = request.recipe.check_for_security_warnings();
+    if has_security_warnings {
+        return Err(ErrorResponse {
+            message: "This recipe contains hidden characters that will be ignored for your safety, as they could be used for malicious purposes.".to_string(),
```

Comment:
isn't it more the case that we are refusing the save this recipe? so say something like, this contains hidden characters, please remove before trying to save?